aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -502,6 +509,76 @@ Tensor& inverse_out(Tensor &result, const Tensor &self) {
   return result;
 }
 
+// This is a type dispatching helper function for 'apply_inverse'
+Tensor& _linalg_inv_out_helper_cpu(Tensor &result, Tensor& infos_lu, Tensor& infos_getri) {
+  // This function calculates the inverse matrix in-place
+  // result should be in column major order and contain matrices to invert
+  // the content of result is overriden by 'apply_inverse'
+  // std::vector<int64_t> infos(batchCount(result), 0);
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(result.scalar_type(), "linalg_inv_out_cpu", [&]{
+    apply_inverse<scalar_t>(result, infos_lu, infos_getri);
+  });
+  return result;
+}
+
+// Computes the inverse matrix of 'input', it is is saved to 'result' in-place
+// LAPACK/MAGMA/cuSOLVER error codes are saved in 'infos' tensors, they are not checked here
+static Tensor& linalg_inv_out_info(Tensor& result, Tensor& infos_lu, Tensor& infos_getri, const Tensor& input) {
+  squareCheckInputs(input);
+  TORCH_CHECK(infos_lu.scalar_type() == kInt,
+    "infos_lu dtype ", infos_lu.scalar_type(), " does not match the expected dtype ", kInt);
+  TORCH_CHECK(infos_getri.scalar_type() == kInt,
+    "infos_getri dtype ", infos_getri.scalar_type(), " does not match the expected dtype ", kInt);
+  TORCH_CHECK(result.scalar_type() == input.scalar_type(),
```

Comment:
We should also add a check that the tensors are on the correct devices (like that the result tensor is on the same device as input)